pkg/model/deployment.go

```diff
@@ -98,6 +98,21 @@ func (d *Deployment) StageMap() map[string]*PipelineStage {
 	return stage
 }
 
+// Stage returns the stage by stage id.
+func (d *Deployment) Stage(id string) (*PipelineStage, bool) {
+	for _, s := range d.Stages {
+		if s.Id == id {
+			return s, true
```

Comment:
nits, should return something like `fmt.Errorf("stage with id (%s) is not available", id)` instead to avoid checking bool flag on the caller side